test/integration_test.tsx

```diff
@@ -416,54 +421,26 @@ function addInteractiveTests(render: typeof mount) {
   });
 }
 
+const createDefaultAdapter = () => new Adapter();
+function setAdapter(createNewAdapter: () => Adapter) {
+  beforeEach(() => {
+    configure({ adapter: createNewAdapter() });
+  });
+
+  afterEach(() => {
+    configure({ adapter: createDefaultAdapter() });
+  });
+}
+
 describe('integration tests', () => {
   before(() => {
-    configure({ adapter: new Adapter() });
+    configure({ adapter: createDefaultAdapter() });
   });
 
   describe('"mount" rendering', () => {
     addStaticTests(mount);
     addInteractiveTests(mount);
 
-    it('supports simulating events on deep Components and elements', () => {
```

Comment:
I just moved this test into its own describe block that uses the `setupAdapter` helper since passing the `adapter` option to `mount` or `shallow` doesn't change the adapter used for all methods (e.g. `.debug`)

Reply:
Is this a bug/missing feature in Enzyme (using the `adapter` passed to `mount` in some cases but the global adapter in others)?

Reply:
Yea, it seems to be a bug. Here is [the line where I ran into that issue](https://github.com/enzymejs/enzyme/blob/9229adac2c3b1c4eb8dddab3104302c8cc9f22e2/packages/enzyme/src/Debug.js#L89) if you are curious. May open a PR if I have time.

But I think that changing the adapter using `configure` more closely matches the setup our users do so I think it makes the test a little more reliable